src/debt_order/index.ts

```diff
@@ -14,16 +21,33 @@ import { BLOCK_TIME_ESTIMATE_SECONDS } from "../../utils/constants";
 
 export class DebtOrder {
     private debtOrderData: DebtOrderData = {};
+    private debtOrderParams?: DebtOrderParams;
+
+    constructor(private dharma: Dharma) {}
+
+    public async open(params: DebtOrderParams) {
+        const { principal, collateral, interestRate, term, debtorAddress } = params;
+
+        const loanOrder: CollateralizedSimpleInterestLoanOrder = {
+            principalAmount: principal.rawAmount,
+            principalTokenSymbol: principal.tokenSymbol,
+
+            interestRate: interestRate.raw,
+            amortizationUnit: term.unit,
+            termLength: term.length(),
```

Comment:
I must have missed this in the discussion around `term`, but I think this should be `getLength()` -- length() is a little confusing to me.

Reply:
💯 I think we should also do a rev on these intermediary types to ensure we have nice APIs here as well. I'm just ripping on them without much thought.